src/clients/api/mutations/useApproveVrt.ts

```diff
@@ -0,0 +1,28 @@
+import { useMutation, MutationObserverOptions } from 'react-query';
+
+import { approveVrt, IApproveVrtInput, ApproveVrtOutput } from 'clients/api';
+import FunctionKey from 'constants/functionKey';
+import { useTokenContract } from 'clients/contracts/hooks';
+
+const useApproveVrt = (
+  options?: MutationObserverOptions<
+    ApproveVrtOutput,
+    Error,
+    Omit<IApproveVrtInput, 'vrtTokenContract'>
+  >,
+) => {
+  const vrtTokenContract = useTokenContract('vrt');
+  return useMutation(
+    FunctionKey.APPROVE_VRT,
+    params =>
+      approveVrt({
+        vrtTokenContract,
+        ...params,
+      }),
+    {
+      ...options,
+    },
+  );
+};
+
+export default useApproveVrt;
```

Comment:
Can't we use `useApproveToken` directly?

Reply:
We could. The difference is the contract it uses